### PR TITLE
Update Hero Availability & Badge style

### PR DESCRIPTION
**Pull request by hqhhuang**
Bug/issue #, if applicable: 122130081

## Summary
Update style of Availability in the Hero and Badges.

## Dependencies
NA

## Testing
Manual testing, check:
1. badges in the navigator, nav bar, and the availability pill
2. availability pills in the hero section

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [ ] Updated documentation if necessary

### CODE REVIEW

**Comment by hqhhuang** [comment]
@swift-ci test

File: src/components/Badge.vue [hunk 1]
```diff
@@ -53,20 +54,18 @@ export default {
 
   @include font-styles(badge);
   display: inline-block;
-  padding: 2px 10px;
+  padding: 2px 4px;
   white-space: nowrap;
-  background: none;
-  border-radius: var(--badge-border-radius, calc(#{$border-radius} - 1px));
-  border-style: var(--badge-border-style, solid);
```
Comment (mportiz08) [inline src/components/Badge.vue:60]: Should we keep these and just change the defaults so that users can continue to theme the style/width of the badge border?

File: src/components/DocumentationTopic/Summary/Availability.vue [hunk 2]
```diff
@@ -99,16 +97,23 @@ export default {
...
   gap: 10px;
-  margin-top: 25px;
-}
-
-.badge {
-  margin: 0;
+  margin-top: rem(15px);
+  @include font-styles(body-reduced);
+
+  div:not(:last-child) {
```
Comment (mportiz08) [inline src/components/DocumentationTopic/Summary/Availability.vue:109]: These selectors are a little messy and hard to follow. I have one idea on a possible way to make it clearer that may or may not work, but it might be worth looking into.

1. Add the same class to every span that you want to have the separator (can be different components)
2. Change the rule to simply be `.that-class + .that-class` (every one preceded by another and apply the left border—no need for using not or last-child, etc)

I may be misunderstanding what this is doing, so feel free to ignore if that's the case.

File: src/components/DocumentationTopic/Summary/Availability.vue [hunk 3]
```diff
@@ -119,72 +124,43 @@ export default {
...
-  }
-}
-
-.deprecated {
-  color: var(--color-badge-deprecated);
-
-  @include prefers-dark {
-    color: var(--color-badge-dark-deprecated);
-  }
+  fill: var(--color-figure-gray) !important;
```
Comment (mportiz08) [inline src/components/DocumentationTopic/Summary/Availability.vue:127]: Can we do this in a way that doesn't need `!important`?

Haven't tried these, but these are the workarounds I might try first:
* make sure your override has the right precedence
* possibly override `--color-svg-icon` here instead

File: src/components/Badge.vue [hunk 4]
```diff
@@ -53,20 +54,18 @@ export default {
...
-  color: var(--badge-color);
-
-  .theme-dark & {
-    --badge-color: var(--badge-dark-color);
+  border-radius: var(--badge-border-radius, $border-radius);
+  border: none;
+  margin: auto;
+  margin-left: 5px;
+  color: var(--colors-button-text, var(--color-button-text));
+  background-color: var(--badge-color);
```
Comment (mportiz08) [inline src/components/Badge.vue:64]: Now that we're using the badge color as a background fill instead of a border, we'll likely need to re-think the color values themselves, since I don't think the contrast ratio will be high enough in terms of accessibility.

File: src/components/Badge.vue (hunk 1 above)
Comment (hqhhuang) [reply src/components/Badge.vue:60]: Good point!

File: src/components/DocumentationTopic/Summary/Availability.vue (hunk 2 above)
Comment (hqhhuang) [reply src/components/DocumentationTopic/Summary/Availability.vue:109]: Using sibling selector and apply left border would cause the dividing line to appear on items on a new line when the container wraps. I couldn't find a clean way to not apply border style to first element in each row. Having a dividing line on the right of the last item in a row looks not as broken.
But I added a new class name & cleaned this up a little. Let me know what you think.

**Comment by hqhhuang** [comment]
@swift-ci test

File: src/components/DocumentationTopic/Summary/Availability.vue (hunk 2 above)
Comment (mportiz08) [reply src/components/DocumentationTopic/Summary/Availability.vue]: Understand why you need to do the right border for wrapping 👍

How about something like this maybe?

```diff
diff --git a/src/components/DocumentationTopic/Summary/Availability.vue b/src/components/DocumentationTopic/Summary/Availability.vue
index 29f8156b..e1079bb5 100644
--- a/src/components/DocumentationTopic/Summary/Availability.vue
+++ b/src/components/DocumentationTopic/Summary/Availability.vue
@@ -10,18 +10,18 @@
 
 <template>
   <Section class="availability" role="complementary" :aria-label="$t('sections.availability')">
-    <div
+    <span
       v-for="technology in technologies"
       class="technology"
       :key="technology"
     >
       <TechnologyIcon class="tech-icon" />
-      <span class="availability-info">{{ technology }}</span>
-    </div>
+      <span>{{ technology }}</span>
+    </span>
 
-    <div
+    <span
       v-for="platform in platforms"
-      class="platform availability-info"
+      class="platform"
       :class="changesClassesFor(platform.name)"
       :key="platform.name"
     >
@@ -32,7 +32,7 @@
       />
       <Badge v-if="platform.deprecatedAt" variant="deprecated" />
       <Badge v-else-if="platform.beta" variant="beta" />
-    </div>
+    </span>
   </Section>
 </template>
 
@@ -106,11 +106,14 @@ $icon-size-default: 20px;
   margin-top: rem(15px);
   @include font-styles(body-reduced);
 
-  &:not(:last-child) .availability-info::after {
-    // apply on the span of technology class to ensure same separator height
-    content: " ";
-    border-right: 1px solid;
-    padding-left: 10px;
+  & > * {
+    border-right: 1px solid currentColor;
+    padding-right: 10px;
+
+    &:last-child {
+      border-right: none;
+      padding-right: 0;
+    }
   }
 }
 ```

Comment (mportiz08) [reply src/components/DocumentationTopic/Summary/Availability.vue]: And bonus points if you use a css variable for the `10px` since it would be used for the `gap` and also this padding.

**Comment by hqhhuang** [comment]
@swift-ci test

File: src/components/DocumentationTopic/Summary/Availability.vue (hunk 2 above)
Comment (marinaaisa) [reply src/components/DocumentationTopic/Summary/Availability.vue]: I haven't tried Marcus's proposal but it could work. I'd add a small detail: we could revert this change too if we don't use a `content` property https://github.com/apple/swift-docc-render/pull/783/files#diff-61f11105739958fab994f705ecdd503c38e30566ec60f517046ff58c2298d9afL150

File: src/components/DocumentationTopic/Summary/Availability.vue [hunk 5]
```diff
@@ -10,18 +10,18 @@
 
 <template>
   <Section class="availability" role="complementary" :aria-label="$t('sections.availability')">
-    <Badge
+    <div
```
Comment (marinaaisa) [inline src/components/DocumentationTopic/Summary/Availability.vue:13]: Instead of rendering a `div` , it could be more semantic if we render `li` elements inside a `ul`.

**Review by marinaaisa — CHANGES_REQUESTED** [review]
I tested the new style and it looks great 🚀 

Marcus' comments to simplify the CSS were very reasonable, let's see if we can achieve it.
I left a comment to improve the AX semantics.

File: src/components/DocumentationTopic/Summary/Availability.vue (hunk 5 above)
Comment (mportiz08) [reply src/components/DocumentationTopic/Summary/Availability.vue:13]: Maybe. I would definitely use a span or make sure to use inline or inline-block list items instead of a div block since these are all inline text that are meant to live on the same line(s).

File: src/components/DocumentationTopic/Summary/Availability.vue [hunk 6]
```diff
@@ -99,92 +97,77 @@ export default {
...
-  &-modified {
-    border-color: var(--color-changes-modified);
+.availability {
+  display: flex;
+  flex-flow: row wrap;
+  gap: $availability-info-spacing;
+  margin-top: rem(15px);
+
+  & > * {
+    @include font-styles(body-reduced);
```
Comment (mportiz08) [inline src/components/DocumentationTopic/Summary/Availability.vue:158]: You can probably lift this up into the parent since it's probably redundant to be setting the font styles for every item

File: src/components/DocumentationTopic/Summary/Availability.vue [hunk 7]
```diff
@@ -99,92 +97,77 @@ export default {
...
+    &::after {
+      content: '';
+      display: inline-block;
+      width: 1px;
+      height: 14px;
```
Comment (mportiz08) [inline src/components/DocumentationTopic/Summary/Availability.vue:163]: Is this meant to match the font size or line height? You may want to consider using em units if that's the case.

File: src/components/DocumentationTopic/Summary/Availability.vue [hunk 8]
```diff
@@ -99,92 +97,77 @@ export default {
...
+      background: black;
```
Comment (mportiz08) [inline src/components/DocumentationTopic/Summary/Availability.vue:164]: This should be currentColor or something to switch between black/white depending on the color scheme, right? Or is it purposefully always black?

File: src/components/DocumentationTopic/Summary/Availability.vue (hunk 5 above)
Comment (hqhhuang) [reply src/components/DocumentationTopic/Summary/Availability.vue]: I changed it to use <span> for now.
I have a bug tracking doing a cleanup of the `Summary` component/folder, since we no longer really have a summary section. Maybe as a part of that I can look into treating availability info as a `ul`.

File: tests/unit/components/DocumentationTopic/Summary/Availability.spec.js [hunk 9]
```diff
@@ -74,52 +75,45 @@ describe('Availability', () => {
     expect(section.attributes('role')).toBe('complementary');
   });
 
-  it('renders a `Badge` for technologies, a `Badge` and `AvailabilityRange` for each platform', () => {
+  it('renders a `div` for technologies, a `AvailabilityRange` and a `Badge` if applicable for each platform', () => {
```
Comment (mportiz08) [inline tests/unit/components/DocumentationTopic/Summary/Availability.spec.js:78]:
```suggestion
  it('renders a `span` for technologies, a `AvailabilityRange` and a `Badge` if applicable for each platform', () => {
```

(or maybe update the whole test name to reflect how it works today in case it's different)

File: src/components/DocumentationTopic/Summary/Availability.vue [hunk 10]
```diff
@@ -99,92 +97,77 @@ export default {
...
+  gap: $availability-info-spacing;
+  margin-top: rem(15px);
+  @include font-styles(body-reduced);
+
+  & > * {
+    &::after {
+      content: '';
+      display: inline-block;
+      width: 1px;
+      height: rem(14px);
```
Comment (mportiz08) [inline src/components/DocumentationTopic/Summary/Availability.vue:163]: Would `1em` work here for the intended effect? Not super important, just curious.

**Review by mportiz08 — APPROVED** [review]
LGTM now

**Comment by hqhhuang** [comment]
@swift-ci test

File: src/components/DocumentationTopic/Summary/Availability.vue (hunk 5 above)
Comment (marinaaisa) [reply src/components/DocumentationTopic/Summary/Availability.vue]: Looking forward to that refactor of the Summary component! It's okay if we apply the semantic changes then.

**Review by marinaaisa — APPROVED** [review]
LGTM! Thanks 👍
